DB/src/test/java/io/deephaven/db/v2/utils/TestKeyedArrayBackedMutableTable.java

```diff
@@ -278,10 +267,24 @@ private void waitForCompletion() throws InterruptedException, ExecutionException
         }
     }
 
-    private void handleDelayedRefresh(BaseArrayBackedMutableTable kabut) {
-        new Thread(() -> {
-            SleepUtil.sleep(200);
-            LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(kabut::refresh);
-        }).start();
+    private void handleDelayedRefresh(final BaseArrayBackedMutableTable table, final FunctionalInterfaces.ThrowingRunnable<IOException> action) throws Exception {
+        final Thread refreshThread;
+        final Semaphore gate = new Semaphore(0);
+        table.setOnPendingChange(gate::release);
+        try {
+            refreshThread = new Thread(() -> {
+                gate.acquireUninterruptibly();
+                LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(table::refresh);
+            });
+            refreshThread.start();
+            action.run();
```

Comment:
Maybe we can get rid of the gate, and just have `setOnPendingChange(refreshThread::start)`?

Or, can the action itself be invoking the LTM code:

```
table.setOnPendingChange(() -> LiveTableMonitor.DEFAULT.runWithinUnitTestCycle(table::refresh))
```
?

Reply:
No, that defeats the intentionally-concurrent design of the unit test.